src/components/MultiHopTrade/hooks/useAllowanceApproval/useAllowanceApproval.tsx

```diff
@@ -26,10 +28,19 @@ export const useAllowanceApproval = (
     return _executeAllowanceApproval()
   }, [_executeAllowanceApproval, stopPollingBuildApprovalTx, stopPollingIsApprovalNeeded])
 
+  useEffect(() => {
+    if (isApprovalNeeded !== undefined && isLoading) {
+      setIsLoading(false)
+    } else if (isApprovalNeeded === undefined && !isLoading) {
+      setIsLoading(true)
+    }
+  }, [isApprovalNeeded, isLoading])
```

Comment:
thought this was clearer than the UI checking `isApprovalNeeded === undefined`

Reply:
If it's not loading... set is loading?

<img src="https://media4.giphy.com/media/ghuvaCOI6GOoTX0RmH/giphy.gif"/>

Reply:
handled with comment :)